intake/source/base.py

```diff
@@ -215,6 +231,42 @@ def to_spark(self):
         """
         raise NotImplementedError
 
+    @property
+    def entry(self):
+        if self._entry is None:
+            raise NoEntry("Source was not made from a catalog entry")
+        return self._entry
+
+    def configure(self, **kwargs):
```

Comment:
This name strongly suggests to me that it is mutating the current instance, rather than returning a new instance. Could you live with `duplicate(...)`?

Reply:
Duplicate isn't right either, because the new one is not the same as the old one. 
Other suggestions? `change`, `copy_and_configure`? I don't yet know if the old `get`/`__call__` will be deprecated, so perhaps this is not the ones users will actually type.

Reply:
That's true. I have the same issue with `change` that I have with `configure`. I think `copy_and_configure` is pretty good. Other ideas, maybe not better than `copy_and_configure`:

* `new_instance(...)`
* `variation(...)`
* `new_variation(...)`
* `with_params(...)`

Reply:
`configure_new` ?